### PR TITLE
Switch to async/await.

### PR DESCRIPTION
**Pull request by davidlehn**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -59,171 +60,134 @@ bedrock.events.on('bedrock.ready', callback => {
...
-      }));
-      callback();
-    }],
-    beforeStatic: ['logger', callback => bedrock.events.emit(
-      'bedrock-express.configure.static', server, callback)],
-    static: ['beforeStatic', (callback, results) => {
-      if(results.beforeStatic === false) {
-        return callback();
+async function init() {
+  const emit = promisify(bedrock.events.emit);
```
Comment (dlongley) [inline lib/index.js]: `bedrock.events.emit` already returns a promise ... so this shouldn't be necessary. Did you find that it was?

Comment (davidlehn) [reply lib/index.js]: Without `promisify` it's accessing keys of `undefined` and throwing unhandled promise rejections.  I figured there was some calling convention confusion related to callbacks and this fixed it.  I didn't explore further.

Comment (dlongley) [reply lib/index.js]: @davidlehn, we should fix that in bedrock... sorry, another yak to shave. I guess we can accept this for now but add an issue to remove it later once the bug(s) in bedrock are fixed for this method.

File: package.json [hunk 2]
```diff
@@ -33,14 +33,21 @@
...
   },
   "peerDependencies": {
     "bedrock": "^1.5.0",
     "bedrock-server": "^2.0.0"
   },
   "directories": {
     "lib": "./lib"
+  },
+  "devDependencies": {
+    "eslint": "^5.13.0",
```
Comment (mattcollier) [inline package.json]: this is technically unncecessary since eslint-config-digitalbazaar provides its own eslint dep.

Possible exception would be if one is going to setup `npm run lint` script as is being done elsewhere.

Comment (davidlehn) [reply package.json]: I did setup a lint script, so figured eslint should be a direct dependency.

Comment (mattcollier) [reply package.json]: Yeah, I totally get the reasoning behind this, but it's going to be sad when we have to go to eslint.next and change this in all the places.  Seems to me that it would be safe to use the guarantees of eslint being installed provided by eslint-config-digitalbazaar. 

Also, I have no idea what happens if eslint-config-digitalbazaar specifies one eslint version and packages are specifying another.  Seems easy enough to avoid that complication.

Comment (davidlehn) [reply package.json]: I think we should be more correct with this in case you are using tools like `pnpm` which defaults to a tree vs flat `node_modules`. In that case you need the dependency else it fails to find the `eslint` binary.

Comment (mattcollier) [reply package.json]: @davidlehn in the interest of science, I install pnpm and have found that `pnpm run eslint`on this branch produces the following error.  Are you having different results?
```
bedrock-express@2.1.1-0 lint /home/matt/dev/bedrock-dev/bedrock-express
> eslint 'lib/*.js'

internal/modules/cjs/loader.js:583
    throw err;
    ^

Error: Cannot find module 'prelude-ls'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/home/matt/dev/bedrock-dev/bedrock-express/node_modules/.registry.npmjs.org/eslint/5.13.0/node_modules/optionator/lib/index.js:5:10)
    at Object.<anonymous> (/home/matt/dev/bedrock-dev/bedrock-express/node_modules/.registry.npmjs.org/eslint/5.13.0/node_modules/optionator/lib/index.js:465:4)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
```

Comment (davidlehn) [reply package.json]: Worked fine here with:
pnpm 2.25.6
node v10.15.0
npm 6.7.0

Comment (mattcollier) [reply package.json]: You already have eslint on your path by chance?

Comment (davidlehn) [reply package.json]: Huh, yeah.  If I run the installed eslint direct it fails like you say.  Let's take the conversation off of this issue.

File: lib/index.js (hunk 1 above)
Comment (mattcollier) [reply lib/index.js]: Fix for the `emit` issue here landed in bedrock@1.17.0.

**Comment by davidlehn** [comment]
I think this is ok to merge and release now.  There's an edge case with errors but it's a general Promise listener issue:
https://github.com/digitalbazaar/async-node-events/issues/1

**Comment by mattcollier** [comment]
@davidlehn of course, there is no test suite here.  Have you plugged this into some top-level modules to test for breakage?

**Comment by davidlehn** [comment]
@mattcollier It worked with a couple projects.  I'm not sure how to best test that entire initialization flow.  It is a major version change so we can likely fix problems as we go.  The async/await transformation was straightforward so bugs are more likely typos rather than serious design issues.

I'm not sure what to do with the `async-node-events` bug.  On second thought, that should perhaps be solved before releasing this.  Any errors in the `bedrock.start` function (and all the events it emits) don't cause a quick program failure like before but instead things can keep starting up (in a broken state no doubt).  That's a problem if you reasonably expect errors to be fatal.  That lib has tricky code if anyone wants to take a look.
